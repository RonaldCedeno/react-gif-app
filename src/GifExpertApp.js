import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({defaultCategories = ['Marvel']}) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2 className='initial-title'>Gif Search</h2>
            <AddCategory setCategories={setCategories}/>
            <hr />

            <ol>
                {
                    categories.map( category => (
                        <GifGrid 
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>
        </>
    )
    
}

export default GifExpertApp;