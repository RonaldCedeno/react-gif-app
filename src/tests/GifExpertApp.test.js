import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Prueba de GifExpertApp', () => {

    test('debe de mostrar correctamente el componente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['Marvel', 'Goku'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })

})