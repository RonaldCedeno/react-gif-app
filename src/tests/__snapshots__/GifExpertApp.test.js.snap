// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifExpertApp debe de mostrar correctamente el componente 1`] = `
<Fragment>
  <h2
    className="initial-title"
  >
    Gif Search
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Marvel"
      key="Marvel"
    />
  </ol>
</Fragment>
`;

exports[`Prueba de GifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2
    className="initial-title"
  >
    Gif Search
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Marvel"
      key="Marvel"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;
